src/AppInstallerCLICore/Workflows/ConfigurationFlow.cpp

```diff
@@ -1064,7 +1064,7 @@ namespace AppInstaller::CLI::Workflow
             }
 
             auto promptString = m_isApply ? Resource::String::ConfigurationWarningPromptApply : Resource::String::ConfigurationWarningPromptTest;
-            if (!context.Reporter.PromptForBoolResponse(promptString, Reporter::Level::Warning))
+            if (!context.Reporter.PromptForBoolResponse(promptString, ReporterLevel::Warning))

```

Comment:
This brings up an issue; if you have disabled `Warning` and we prompt at `Warning`, are we just going to sit waiting for user input without having actually issued a prompt on the console?

Maybe prompts need to enable all levels while they are active?

Alternatively, prompts at disabled levels could simply apply their non-interactive choice; after all, the user requested no warnings, and this prompt is a warning...

Reply:
I like the flow of appearing as their non-interactive choice; I think it makes for a cleaner flow. Will update

Reply:
That might also require checking all of the prompts for the levels that they output at.  For instance, `InstallerAbortsTerminalPrompt` uses warnings by default and thus would need to check the warning level.

Reply:
> That might also require checking all of the prompts for the levels that they output at. For instance, `InstallerAbortsTerminalPrompt` uses warnings by default and thus would need to check the warning level.

It would require going through all the prompts and updating either way - either to re-enable the warning level temporarily, or to make it non-interactive.

The other approach could be to add an optional boolean to `Reporter.Warn()`, `Reporter.Info`, etc. for bypassing suppression

Reply:
Or maybe expose the m_isEnabled in OutputStream class, then within the PromptFor* method implementations, after GetOutputStream(level), we can check if it's enabled and act accordingly.

Reply:
> Or maybe expose the m_isEnabled in OutputStream class, then within the PromptFor* method implementations, after GetOutputStream(level), we can check if it's enabled and act accordingly.

I tried that actually, but every time `Reporter.Warn()` is called it goes through `GetBasicOutputStream` it means that there is not a persistent stream for any of the reporter levels and so `m_enabled` doesn't actually have any effect. This is probably a flaw in the implementation, but not one that I know how to fix (as evidenced by all my troubles even using the simple enum as flags)

Reply:
Maybe we are not looking at the same place? I was referring to [here](https://github.com/microsoft/winget-cli/blob/464e177f3b355ffef3d9f96fdebed9004407d181/src/AppInstallerCLICore/ExecutionReporter.cpp#L173), within  PromptForBool, PromptForEnter, PromptForPath implementation, after we get the stream, we can check if the stream is enabled(.m_enabled to be exposed) and act accordingly. So we don't need to go over every calling place.

Reply:
> Maybe we are not looking at the same place? I was referring to [here](https://github.com/microsoft/winget-cli/blob/464e177f3b355ffef3d9f96fdebed9004407d181/src/AppInstallerCLICore/ExecutionReporter.cpp#L173), within PromptForBool, PromptForEnter, PromptForPath implementation, after we get the stream, we can check if the stream is enabled(.m_enabled to be exposed) and act accordingly. So we don't need to go over every calling place.

Ah, I see. I thought you were talking about setting m_enabled to be false for the stream instead of using the flags. 

Yes, I think it would be easy enough to expose m_enabled and check it

Reply:
After looking at the code, I thought it would be best to make the output stream check at the `IsInteractivityAllowed` level to take advantage of the already existing behaviors for when interactivity is not allowed. Since all of the `PromptFor` methods had a `Reporter::Level` defined, I removed the default of `Reporter::Level::Info` to make it more clear which level each prompt is at, and added that same level to the Interactivity check. Strictly speaking, there isn't the ability to disable the `Reporter::Level::Info` currently, but wanted to be sure that it wouldn't be missed if anyone goes to update the behavior in the future.